extending-valtimo/plugin/custom-plugin-definition.md

```diff
@@ -82,6 +82,57 @@ class TwitterPluginFactory(
 }
 ```
 
+### Plugin categories
+There are use-cases where you would like to use another plugin in order to share functionality and/or configuration.
+Plugin Categories to the rescue: this functionality lets you choose a plugin implementing a specific interface into your plugin as a `@PluginProperty` in the configuration screen.
+
+For example, you might want to use a supplier for your tweets:
+```kotlin
+@PluginCategory()
+interface TweetSupplier {
+    fun supplyMessage() : String
+}
+```
+
+```kotlin
```

Comment:
At least one implementation of the plugin is required. In this case the PropertyTweetSupplier implements the interface TweetSupplier and supports all required functionality. When searching for configurations for category x all stored PropertyTweetSupplier configurations are found.